packages/editor/src/panels/hierarchy/hierarchynode.tsx

```diff
@@ -126,6 +129,8 @@ export default function HierarchyTreeNode(props: ListChildComponentProps<undefin
   const { setMenu } = useHierarchyTreeContextMenu()
   const renameRef = useRef<HTMLDivElement>(null)
   let isRenameOpen = useState(false)
+  const canSaveChanges = useState(false)
+  const permissionVerified = useState(false)
```

Comment:
i believe we can do this skipping the usage of useState.

we can use the `useFind` feathers hook to get the project permission and then use calculate the boolean value using `userHasProject Permission`

Reply:
I decided to not use that hook and use `await API.instance.service(projectPermissionPath).find` instead because we only check project permissions for prefabs/nodes that belong to a different project than the scene, not for all nodes. This should reduce the number of API calls, making it more cost-effective

Reply:
okay.